pumpkin-lib/src/propagators/cumulative/utils/sparse_set.rs

```diff
@@ -98,12 +104,34 @@ impl<T> SparseSet<T> {
         if self.indices[(self.mapping)(to_remove)] < self.size {
             // The element is part of the domain and should be removed
             self.size -= 1;
-            self.swap(self.indices[(self.mapping)(to_remove)], self.size);
+            self.swap(
+                self.indices[(self.mapping)(to_remove)],
+                self.domain.len() - 1,
+            );
+            let _ = self.domain.pop().expect("Has to have something to pop.");
+            self.indices[(self.mapping)(to_remove)] = usize::MAX;
+        } else if self.indices[(self.mapping)(to_remove)] < self.domain.len() {
+            self.swap(
+                self.indices[(self.mapping)(to_remove)],
+                self.domain.len() - 1,
+            );
             let _ = self.domain.pop().expect("Has to have something to pop.");
             self.indices[(self.mapping)(to_remove)] = usize::MAX;
         }
     }
 
+    pub(crate) fn remove_temporarily(&mut self, to_remove: &T) {
```

Comment:
What does `temporarily` mean?

Reply:
Due to the changes in the semantic minimisation, we have 2 approaches for removing from the sparse-set; 1) fully remove the element from the domain, 2) remove an element from the domain by moving it to the end of the list and then decrementing a size counter. The latter I now refer to as a temporary removal since it can easily be undone (though I quite dislike the former manner of removal and I am planning to rework it in the future)